modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,30 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+TEST(Layer_Reshape_Test, Accuracy)
+{
+    LayerParams lp;
+    lp.type = "Reshape";
+    lp.name = "ReshapeLayer";
+    lp.set("axis", 0); // Set axis to 0 to start reshaping from the first dimension
+    lp.set("num_axes", -1); // Set num_axes to -1 to indicate all following axes are included in the reshape
+    int newShape[] = {1};
+    lp.set("dim", DictValue::arrayInt(newShape, 1));
+
+    Ptr<ReshapeLayer> layer = ReshapeLayer::create(lp);
+
+    std::vector<int> input_shape = {0};
+
+    Mat input(0, input_shape.data(), CV_32F);
```

Comment:
@vpisarev Why do we need to create and pass `input_shape` array for 0D "Mat"? That behavior is not convenient.

---

Anyway we don't need to use `std::vector` overkill here. Regular C arrays works well.